bottle_app.py

```diff
@@ -629,14 +629,59 @@ def api_get_movie_trackpoints():
         labels         = sorted( ( tp['label'] for tp in trackpoint_dicts) )
         frame_dicts    = defaultdict(dict)
 
+        ruler0_x, ruler20_x = -1, -1
+        ruler0_y, ruler20_y = -1, -1
+        ruler0_frame_number = -1
+        sensor_width_mm = 36  # Sensor width 36 millimeters (this is a common value,
+        # but should be adjusted according to actual circumstances)
+        actual_distance_mm = 20
+        mm_per_inch = 25.4  # common value
+
         for tp in trackpoint_dicts:
-            frame_dicts[tp['frame_number']][tp['label']+' x'] = tp['x']
-            frame_dicts[tp['frame_number']][tp['label']+' y'] = tp['y']
+            frame_dicts[tp['frame_number']][tp['label'] + ' x'] = tp['x']
+            frame_dicts[tp['frame_number']][tp['label'] + ' y'] = tp['y']
+            if tp['label'] == 'ruller 0 mm':
+                ruler0_x = tp['x']
+                ruler0_y = tp['y']
+                ruler0_frame_number = tp['frame_number']
+            if tp['label'] == 'ruller 20 mm':
+                ruler20_x = tp['x']
+                ruler20_y = tp['y']
```

Comment:
The function that you create will take the distance of the two ruler points as a parameter. It might be 20 mm, but it might be something else. We will then parse the labels of the trackpoints ending with "mm" to determine how far apart they are.

Reply:
In any case, is that a typo in line 647 -- ruller rather than ruler?